From 87cc4cae8d0d28692193471b9ae77378bcc147ae Mon Sep 17 00:00:00 2001
From: Yaroslav Furman <yaro330@gmail.com>
Date: Sun, 26 May 2019 14:25:44 +0300
Subject: [PATCH 004/157] techpack: audio: pinctrl-lpi: initialise at
 late_initcall

Fixes booting with techpack online and modules disabled.
Without this it spits out a nasty NULL pointer dereference.

Signed-off-by: Yaroslav Furman <yaro330@gmail.com>
Change-Id: Ib51f186ccff1c7f7c7aed27d4ef8319561b3dba4
---
 techpack/audio/soc/pinctrl-lpi.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/techpack/audio/soc/pinctrl-lpi.c b/techpack/audio/soc/pinctrl-lpi.c
index ec075e828bc1..7383d3282cdf 100644
--- a/techpack/audio/soc/pinctrl-lpi.c
+++ b/techpack/audio/soc/pinctrl-lpi.c
@@ -956,7 +956,17 @@ static struct platform_driver lpi_pinctrl_driver = {
 	.remove = lpi_pinctrl_remove,
 };
 
-module_platform_driver(lpi_pinctrl_driver);
+static int __init lpi_init(void)
+{
+	return platform_driver_register(&lpi_pinctrl_driver);
+}
+late_initcall(lpi_init);
+
+static void __exit lpi_exit(void)
+{
+	platform_driver_unregister(&lpi_pinctrl_driver);
+}
+module_exit(lpi_exit);
 
 MODULE_DESCRIPTION("QTI LPI GPIO pin control driver");
 MODULE_LICENSE("GPL v2");
-- 
2.48.1

